### PR TITLE
feat: `Actor`, `AIProcess`, `ActorPackage` RE

### PR DESCRIPTION
**Pull request by lStewieAl**
Adds AIProcess struct, the fields line up with CommonLibF4.

### CODE REVIEW

**Comment by Qudix** [comment]
It seems like you forgot to move `ActorPackage` and `AIProcess` into their own files?

**Comment by lStewieAl** [comment]
> It seems like you forgot to move `ActorPackage` and `AIProcess` into their own files?

I was following how it was done in CommonLibF4, but sure I can separate them

File: CommonLibSF/include/RE/A/AIProcess.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#pragma once
+#include "ActorPackage.h"
```
Comment (ThirdEyeSqueegee) [inline CommonLibSF/include/RE/A/AIProcess.h:2]: Add newlines above and below the `#include` plz

**Comment by ThirdEyeSqueegee** [comment]
Maybe remove the empty .cpp files for now? Can always add them later when there's stuff to define

File: CommonLibSF/include/RE/A/AIProcess.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+	// F8
+	class AIProcess
+	{
+		MiddleLowProcessData*  middleLow;          // 00
+		MiddleHighProcessData* middleHigh;         // 08
+		HighProcessData*       high;               // 10
+		ActorPackage           currentPackage;     // 18
+		float                  hourLastProcessed;  // 48
+		std::uint32_t          unk4C;
```
Comment (powerof3) [inline CommonLibSF/include/RE/A/AIProcess.h:19]: add comments with unk offset?

File: CommonLibSF/include/RE/A/AIProcess.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace RE
+{
+	struct MiddleLowProcessData;
+	struct MiddleHighProcessData;
+	struct HighProcessData;
+
+	// F8
+	class AIProcess
+	{
```
Comment (Qudix) [inline CommonLibSF/include/RE/A/AIProcess.h:13]:
```suggestion
    class AIProcess
    {
    public:
```
Without this members in a class are private by default

**Review by Qudix — DISMISSED** [review]

File: CommonLibSF/include/RE/A/AIProcess.h (hunk 3 above)
Comment (ThirdEyeSqueegee) [reply CommonLibSF/include/RE/A/AIProcess.h:13]: Oh if we're doing this there's a ton of other places in the codebase where redundant access modifiers (and a lot of other redundant stuff) should be removed too

Comment (Qudix) [reply CommonLibSF/include/RE/A/AIProcess.h:13]: It's adding public in this case

**Pull request merged**